### PR TITLE
Annotate ImmutableXXX classes for Nullness Checker

### PR DESCRIPTION
**Pull request by dilraj45**
Handles null-hostility of `ImmutableXXX` classes under com.google.common.collect package.

### CODE REVIEW

File: guava/src/com/google/common/collect/ImmutableEntry.java [hunk 1]
```diff
@@ -26,23 +26,23 @@
 @AnnotatedFor({"nullness"})
 @GwtCompatible(serializable = true)
 class ImmutableEntry<K, V> extends AbstractMapEntry<K, V> implements Serializable {
```
Comment (dilraj45) [inline guava/src/com/google/common/collect/ImmutableEntry.java]: Do we need to restrict the instantiation of ImmutableEntry to `@NonNull` types?

I haven't added annotations to restrict instantiation of ImmutableEntry to `@NonNull` key-value pairs so as to keep it more flexible. There are certain classes like `com.google.common.collect.HashBiMap.BiEntry` and `com.google.common.
collect.LinkedHashMultimap.ValueEntry` in guava code base itself which extends ImmutableEntry and doesn't necessitate the keys and values to be `@NonNull`. We can always restrict its type to `@NonNull` by explicitly specifying the type during instantiation as it is handled in other `ImmutableXXX` classes. Please let me know if I'm missing something and this class needs to be annotated for null hostility as well. Thanks!

Comment (mernst) [reply guava/src/com/google/common/collect/ImmutableEntry.java]: That's an interesting fact, and one I would not have expected.  Thanks for discovering it.  It's worth a comment at the declaration of ImmutableEntry.

**Comment by mernst** [comment]
There are merge conflicts that need to be resolved.

**Comment by mernst** [comment]
Are these complete annotations, just signatures, or just the nullness-hastility specified in `ImmutableCollection`?

**Comment by dilraj45** [comment]
Following are the classes which are missing in the annotated-JDK along with the PR links, requesting changes for the same.
- `java.util.EnumMap` - https://github.com/typetools/checker-framework/pull/2036
- `java.io.InvalidObjectDetection` - https://github.com/typetools/checker-framework/pull/2035

**Comment by mernst** [comment]
I don't understand your comment.  How are those related to this pull request?  Is this pull request dependent on them?  Are they dependent on this pull request?  Please give this information in general.

**Comment by dilraj45** [comment]
I found missing annotations for these classes while annotating `ImmutableXXX`. I added it in the comment just to keep a record.

`ImmutableXXX` are completely annotated and if annotations are correct then we can safely merge these changes.

**Comment by dilraj45** [comment]
@mernst, Please review and let me know if any change is needed. Thanks!

**Pull request closed without merging**